siibra/volumes/parcellationmap.py

```diff
@@ -39,35 +39,72 @@
 from nibabel import Nifti1Image
 from nilearn import image
 import pandas as pd
-from dataclasses import dataclass, asdict
+from dataclasses import dataclass
+import hashlib
+from os import path
+import uuid
 
 if TYPE_CHECKING:
     from ..core.region import Region
 
 
-class ExcessiveArgumentException(ValueError): pass
+class ExcessiveArgumentException(ValueError):
+    pass
 
 
-class InsufficientArgumentException(ValueError): pass
+class InsufficientArgumentException(ValueError):
+    pass
 
 
-class ConflictingArgumentException(ValueError): pass
+class ConflictingArgumentException(ValueError):
+    pass
 
 
-class NonUniqueIndexError(RuntimeError): pass
+class NonUniqueIndexError(RuntimeError):
+    pass
 
 
 @dataclass
-class Assignment:
-    input_structure: int
-    centroid: Union[Tuple[np.ndarray], point.Point] 
-    volume: int
-    fragment: str
-    map_value: np.ndarray
+class MapAssignment:
+    """
+    Encodes the (possibly probablistic) assignment of a spatial structure
+    (point, image mask, or map) to a parcellation map.
+    """
 
+    def __init__(
+        self,
+        input_structure: int,
+        centroid: Union[Tuple[np.ndarray], point.Point],
+        volume: int,
+        fragment: str,
+        map_value: np.ndarray,
+        result: CompareMapsResult = None
+    ):
+        self.input_structur = input_structure
+        self.centroid = centroid
+        self.volume = volume
+        self.fragment = fragment
+        self.map_value = map_value
+        self.result = result or CompareMapsResult()
```

Comment:
this is not necessary, as this is done automagically by `@dataclass` decorator

```suggestion
    input_structure: int,
    centroid: Union[Tuple[np.ndarray], point.Point],
    volume: int,
    fragment: str,
    map_value: np.ndarray,
    result: CompareMapsResult = field(default_factory=lambda: CompareMapsResult())
```